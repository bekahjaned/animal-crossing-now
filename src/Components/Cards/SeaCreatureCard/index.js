import React from "react";
import "../index.css";

// TODO
// Change to a functional component
class SeaCreatureCard extends React.Component {
  render() {
    const { image, name, availability, price } = this.props;

    return (
      <div className="critter">
        <img
          src={require(`../../../data/images/Sea Creatures/${image}.png`)}
          alt="sea creature"
        ></img>
        <p className="critter-name">{name}</p>
        <p className="availability">Available: {availability}</p>
        <p>Price: {price} Bells</p>
      </div>
    );
  }
}

export default SeaCreatureCard;
