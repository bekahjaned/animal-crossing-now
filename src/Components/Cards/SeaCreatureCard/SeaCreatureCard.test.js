import React from "react";
import renderer from "react-test-renderer";

import SeaCreatureCard from "./index";

it("renders a SeaCreatureCard", () => {
  // given
  const image = "Horseshoe Crab";
  const name = "Horseshoe Crab";
  const availability = "9pm - 4am";
  const price = "2,500";

  // when
  const tree = renderer
    .create(
      <SeaCreatureCard
        image={image}
        name={name}
        availability={availability}
        price={price}
      />
    )
    .toJSON();

  // then
  expect(tree).toMatchSnapshot();
});
