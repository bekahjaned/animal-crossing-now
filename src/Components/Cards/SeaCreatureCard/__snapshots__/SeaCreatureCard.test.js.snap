// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a SeaCreatureCard 1`] = `
<div
  className="critter"
>
  <img
    alt="sea creature"
    src="Horseshoe Crab.png"
  />
  <p
    className="critter-name"
  >
    Horseshoe Crab
  </p>
  <p
    className="availability"
  >
    Available: 
    9pm - 4am
  </p>
  <p>
    Price: 
    2,500
     Bells
  </p>
</div>
`;
